pronto/api/entry/references.py

```diff
@@ -186,3 +186,169 @@ def unlink_reference(accession, pub_id):
+    """
+    invalid_pmids, failed_pub_ids = [], []
+    pmid_query = f"""
+        SELECT P.pubmed_id
+        FROM protein2publication P
+        INNER JOIN signature2protein S ON P.protein_acc = S.protein_acc
+        WHERE S.signature_acc IN ({','.join(['%s'] * len(sig_accs))})
+    """
+
+    pg_con = utils.connect_pg(utils.get_pg_url())
+
+    with pg_con.cursor() as pg_cur:
+        pg_cur.execute(pmid_query, sig_accs)
+        while pmid_batch := pg_cur.fetchmany(size=1000):
+            citations, new_citations, pub_ids = {}, {}, []
+            not_in_oracle, in_oracle = check_pmid_in_citations([_[0] for _ in pmid_batch], orc_con)
+
+            if len(not_in_oracle) > 0:
+                with orc_con.cursor() as new_cit_cur:
+                    new_citations = get_citations(new_cit_cur, not_in_oracle)
+                    error = insert_citations(new_cit_cur, new_citations)
+
+                if error:
+                    invalid_pmids.extend(list(new_citations.keys()))
+                else:
+                    pub_ids.extend([new_citations[pmid] for pmid in new_citations])
+
+            with orc_con.cursor() as cit_cur:
+                citations = get_citations(cit_cur, in_oracle)
+                pub_ids.extend(get_unlinked_pub_ids(citations, cit_cur))
+
+            for pub_id in pub_ids:
+                error = relate_entry_to_pubs(entry_acc, pub_id, orc_con)
+                if error:
+                    failed_pub_ids.append(pub_id)
+
+    pg_con.close()
+    orc_con.commit()
+
+    return invalid_pmids, failed_pub_ids
+
+
+def check_pmid_in_citations(pmids: list[int], orc_con) -> tuple[list[int], list[int]]:
+    """Check for pmid citations that are not listed in INTERPRO.CITATION
+
+    :param pmids: list of pubmed ids from postgresql publication table
+    :param orc_con: interpro oracle db connection
+    """
+    placeholder = ', '.join([':' + str(i) for i in range(1, len(pmids) + 1)])
+    citation_query = f"""
+    SELECT PUBMED_ID
+    FROM INTERPRO.CITATION
+    WHERE PUBMED_ID IN ({placeholder})
+    """
+    params = {str(i): pmid for i, pmid in enumerate(pmids, start=1)}
+
+    with orc_con.cursor() as orc_cur:
+        orc_cur.execute(citation_query, params)
+        in_oracle = [row[0] for row in orc_cur]
+        not_in_oracle = set(pmids).difference(set(in_oracle))
+
+    return list(not_in_oracle), in_oracle
+
+
+def get_unlinked_pub_ids(citations: dict, cur) -> list[str]:
+    """Retrieve pub_ids for citations retrieved from api.annotation.get_citations()
+
+    :param citations: dict from api.annotation.get_citations()
+    :param cur: active oracle db cursor
+    """
+    placeholder = ', '.join([':' + str(i) for i in range(1, len(citations.keys()) + 1)])
+    query = f"""
+        SELECT C.PUB_ID
+        FROM INTERPRO.CITATION C
+        WHERE C.PUBMED_ID IN ({placeholder}) 
+            AND C.PUB_ID NOT IN (SELECT PUB_ID FROM SUPPLEMENTARY_REF)
```

Comment:
Explicit the schema of `SUPPLEMENTARY_REF` so the query does not fail when connected with a different user from `INTERPRO`:

```suggestion
          AND C.PUB_ID NOT IN (SELECT PUB_ID FROM INTERPRO.SUPPLEMENTARY_REF)
```